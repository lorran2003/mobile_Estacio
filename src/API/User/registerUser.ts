import { DataUser } from "@/src/type/DataUser";

export interface TypeRegisterUser {
    'email': string;
    'password': string;
    'name': string;
    'phone': string;
    'cpf': string

}


export const registerUser = async (user: TypeRegisterUser) => {
    try {

        const url: URL = new URL('https://localhost:5001/api/User/create-user');

        const response = await fetch(url, {
            method: "POST",
            body: JSON.stringify(user),
            headers: { "Content-type": "application/json" },
            
        });
        const data: string  = await response.json();
        return data
    }
    catch (e) {
        
        throw new Error("Algo deu errado");
        
    }
};
