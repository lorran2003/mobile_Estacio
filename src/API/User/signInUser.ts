import { DataUser } from "@/src/type/DataUser";

export interface SignIn {
    'email': string | null;
    'password': string | null;

}

const email = '123';
const password = '123';

const _user : DataUser = {
    'name': 'Lorran',
    'email': 'lorran@gmail.com',
}

export const signInUser = async (user: SignIn) => {
    try {

        // const url: URL = new URL('http://192.168.1.245:3000/products');

        // const response = await fetch(url, {
        //     method: "GET",
        //     headers: { "Content-type": "application/json" },
        // });
        // const data: PropsAPI[] | undefined = await response.json();

        // return data;

        const inputEmail = user.email;
        const inputPassword = user.password;

        const response = email === inputEmail && password === inputPassword ? _user : false;

        return response;
    }
    catch {

    }
};
