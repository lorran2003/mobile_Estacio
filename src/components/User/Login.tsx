import { View, Text, TouchableOpacity } from 'react-native'
import { TitlePages } from '../TitlePages'
import { Input } from '../Input'
import { Logo } from '../Logo'
import { Link, Redirect } from 'expo-router'
import { useState } from 'react'
import { AntDesign } from '@expo/vector-icons'
import { useSession } from '../AuthContext'

export function Login() {
    const [email, setEmail] = useState<string>('');
    const [password, setPassword] = useState<string>('');
    const { signIn, session, isLoading } = useSession();

    if (session) {

        return <Redirect href={'/(tabs)/user/profile'} />
    }

    return (

        <View className='w-full items-center gap-10'>


            <TitlePages title='Login' />

            <View className='relative w-11/12 py-10 bg-zinc-50 justify-center items-center rounded-md gap-5'>

            {isLoading && (
                <View className='absolute z-10 w-full h-full justify-center items-center bg-zinc-50/80'>
                    <View className='animate-spin'>
                        <AntDesign name="loading2" size={30} color="#CA9D37" />
                    </View>
                </View>
            )}

                <Logo />

                <View className='justify-center items-center gap-3'>

                    <Text className='font-semibold text-lg'>Ja é cliente Dalaza:</Text>

                    <Input
                        label='E-mail ou telefone :'
                        onChangeText={setEmail}
                    />

                    <Input
                        label='Senha :'
                        onChangeText={setPassword}
                    />

                </View>

                <TouchableOpacity
                    className='bg-[#CA9D37] rounded-md py-2 px-8'
                    onPress={() => signIn({ email, password })}
                >
                    <Text className='text-zinc-50 font-semibold text-xl'>Login</Text>
                </TouchableOpacity>

                <Link href={'/user/register'}>
                    <Text className='text-zinc-800/60 underline'>
                        Se não? Registre aqui
                    </Text>
                </Link>
            </View>
        </View>

    )
}