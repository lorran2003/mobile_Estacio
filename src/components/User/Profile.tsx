import { View, Text, Pressable, Image } from 'react-native';
import { TitlePages } from '../TitlePages';
import { Entypo } from '@expo/vector-icons';

export function ProfileUser() {

    const imageDefault = require('@/src/assets/images/imageDefaultUser.jpg');

    const editPhoto = () => {
        
    }

    return (
        <View className='w-full items-center gap-10'>

            <TitlePages title='Meu perfil' />

            <View className='w-full bg-zinc-50 mx-5 py-5 justify-center items-center gap-5'>


                <View className='relative'>
                    <View className='w-40 h-40 rounded-full border border-[#CA9D37] bg-zinc-50'>
                        <Image
                            source={imageDefault}
                            className='w-full h-full rounded-full'
                        />
                    </View>

                    <Pressable
                        className='z-10 absolute rounded-full bg-neutral-200 p-2 bottom-0 right-0'
                        onPress={() => editPhoto()}
                    >
                        <Entypo name="edit" size={24} color="black" />
                    </Pressable>
                </View>

                <Text className='text-zinc-800'>Name</Text>

                <Text className='text-zinc-800'>Email</Text>

                <View className='gap-5'>

                    <Pressable className='bg-[#CA9D37] rounded-md p-2 w-fit' >
                        <Text className='text-zinc-50 text-lg'>Editar Perfil</Text>
                    </Pressable>

                    <Pressable className='bg-neutral-200 rounded-md p-2 items-center justify-center'>
                        <Text>Sair</Text>
                    </Pressable>

                </View>

            </View>

        </View>
    )
}