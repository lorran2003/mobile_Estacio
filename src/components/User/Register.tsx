import { View, Text, Pressable } from 'react-native';
import { TitlePages } from '../TitlePages';
import { Logo } from '../Logo';
import { Input } from '../Input';
import { Link } from 'expo-router';

export function RegisterUser() {
  return (
    <View className='w-full items-center gap-10'>

      <TitlePages title='Registro' />

      <View className='w-11/12 py-5 bg-zinc-50 justify-center items-center rounded-md gap-5'>

        <Logo />

        <Text className='font-semibold text-lg'>Ja é cliente Dalaza:</Text>

        <View className='justify-center items-center gap-3'
        >

          <Input label='Nome completo: ' />

          <Input label='Telefone :' />

          <Input label='E-mail :' />

          <Input label='CPF :' />

          <Input label='Endereço :' />

          <Input label='CEP :' />

          <Input label='Complemento : (opcional)' />

          <Input label='Senha :' />

        </View>

        <Pressable className='bg-[#CA9D37] rounded-md py-2 px-8'>
          <Text className='text-zinc-50 font-semibold text-xl'>Registrar</Text>
        </Pressable>

        <Link href={'/user/login'}>
          <Text className='text-zinc-800/60 underline'>
          Já possui conta? Logue aqui
          </Text>
        </Link>

      </View>

    </View>
  )
}