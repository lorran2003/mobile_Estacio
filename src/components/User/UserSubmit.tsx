import { Href, Link } from 'expo-router'
import React from 'react'
import { Text, Pressable } from 'react-native'

export function UserSubmit({ title, text, router }: { title: string, text: string, router: Href }) {
    return (
        <>
            <Pressable className='bg-[#CA9D37] rounded-md py-2 px-8'>
                <Text className='text-zinc-50 font-semibold text-xl'>{title}</Text>
            </Pressable>

            <Link href={router}>
                <Text className='text-zinc-800/60 underline'>
                    {text}
                </Text>
            </Link> 
        </>
    )
}