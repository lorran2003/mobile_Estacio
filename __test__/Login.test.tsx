import { Login } from '@/src/components/User/Login';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
});