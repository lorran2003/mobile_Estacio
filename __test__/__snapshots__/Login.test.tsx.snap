// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  className="w-full items-center gap-10"
>
  <View
    className="bg-zinc-50 w-full p-6 "
  >
    <Text
      className="text-[#CA9D37] text-4xl font-bold"
    >
      Login
    </Text>
  </View>
  <View
    className="relative w-11/12 py-10 bg-zinc-50 justify-center items-center rounded-md gap-5"
  >
    <View
      className="absolute z-10 w-full h-full justify-center items-center bg-zinc-50/80"
    >
      <View
        className="animate-spin"
      >
        <Text />
      </View>
    </View>
    <View
      className="size-24 items-center justify-center rounded-full border-solid border-[#CA9D37] border"
    >
      <Image
        source={1}
        style={
          {
            "borderRadius": 9999,
            "height": "100%",
            "objectFit": "cover",
            "width": "100%",
          }
        }
      />
    </View>
    <View
      className="justify-center items-center gap-3"
    >
      <Text
        className="font-semibold text-lg"
      >
        Ja é cliente Dalaza:
      </Text>
      <View
        className="gap-2 justify-center items-centers"
      >
        <Text
          className="text-xl opacity-60"
        >
          E-mail ou telefone :
        </Text>
        <TextInput
          className="rounded-lg border border-[#CA9D37] w-72 h-12 p-2"
          cursorColor="#000"
          onChangeText={[Function]}
        />
      </View>
      <View
        className="gap-2 justify-center items-centers"
      >
        <Text
          className="text-xl opacity-60"
        >
          Senha :
        </Text>
        <TextInput
          className="rounded-lg border border-[#CA9D37] w-72 h-12 p-2"
          cursorColor="#000"
          onChangeText={[Function]}
        />
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        className="text-zinc-50 font-semibold text-xl"
      >
        Login
      </Text>
    </View>
    <Text
      href="/user/register"
      onPress={[Function]}
      role="link"
    >
      <Text
        className="text-zinc-800/60 underline"
      >
        Se não? Registre aqui
      </Text>
    </Text>
  </View>
</View>
`;
